File: 0070-climbing-stairs/0070-climbing-stairs.go
class Solution {
 func climbStairs(_ n: Int) -> Int {
        var dp = [1,1]
        if n < 2
        {
            return 1
        }
        if n == 2
        {
            return dp[0] + dp[1]
        }
        
        for i in 2...n
        {
            dp.append(dp[i-1] + dp[i-2])
        }
        
        return dp[n]
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.php
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let n = n as usize;
        let mut dp = vec![1,2];
        
        for i in 2..n {
            dp.push(
                dp[i - 1] + dp[i - 2]
            );
        }
        
        dp[n - 1]
    }
}